File: src/main/java/com/pigutu/app/xforum/model/Post.java
package com.pigutu.app.xforum.model;

import com.pigutu.app.annotation.Column;
import com.pigutu.app.mapper.mybatis.BaseModel;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.apache.ibatis.annotations.Result;

import java.sql.Date;
import java.util.ArrayList;

/**
 *
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Post extends BaseModel {
    public User user = new User();

    public int bar_id = 0; // 1美图 2美图吧 3斗图 4斗图吧
    public Date publish_time = new Date(System.currentTimeMillis());//发布时间
    public int status = 0;//帖子状态 0正常 1 封禁
    public int count_read = 0;
    public int count_like = 0;
    public int count_reply = 0;
    public int count_forward = 0;//转发数量
    public String content = ""; //内容zhe
    public Date last_update_time = new Date(System.currentTimeMillis()); //最后更新时间
    public ArrayList<String> image_list = new ArrayList();//显示的图片，最多9张，暂定
}

File: src/main/java/com/pigutu/app/xforum/model/Post.kt
package com.pigutu.app.xforum.model

import com.pigutu.app.mapper.mybatis.BaseModel
import java.sql.Date
import javax.persistence.Column
import javax.persistence.Transient

/**
 *
 */
data class Post(
        @Column(name = "user_id")
        var user: User = User(),
        var bar_id: Int = 0, // 1美图 2美图吧 3斗图 4斗图吧
        var publish_time: Date = Date(System.currentTimeMillis()), //发布时间
        var status: Int = 0, //帖子状态 0正常 1 封禁
        var count_read: Int = 0,
        var count_like: Int = 0,
        var count_reply: Int = 0,
        var count_forward: Int = 0, //转发数量
        var content: String = "", //内容zhe
        var content_summary: String = "", //内容摘要
        var last_update_time: Date = Date(System.currentTimeMillis()), //最后更新时间
        var image_list: ArrayList<String> = ArrayList()//显示的图片，最多9张，暂定
) : BaseModel()
